src/loop.c

```diff
@@ -91,3 +92,31 @@ int main() {
     loop(epoll_c, events);
   }
 }
+
+void read_message(int file_descriptor) {
+  int message_len = full_message_availiable(file_descriptor);
+
+  if (message_len) {
+    uint8_t message[PIECE_SIZE_BYTES +
+                    8]; // The largest valid message size available
+    uint8_t message_type;
+    // https://pubs.opengroup.org/onlinepubs/007904975/functions/recv.html
+    // Peek the message at the socket.
+    recv(file_descriptor, message, (size_t)MAX_SIZE_MESSAGE_INT * 4, 0);
```

Comment:
less likely to make a mistake and overflow by writing it this way  
```suggestion
    recv(file_descriptor, message, sizeof(message), 0);
```